### PR TITLE
Add job name tag to pipelines in queues

### PR DESCRIPTION
**Pull request by sarah-witt**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?
- builds on https://github.com/jenkinsci/datadog-plugin/pull/73 to support pipelines
- With the addition of support for pipelines, we can now add pipeline tags to queue metrics

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java [hunk 1]
```diff
@@ -85,10 +88,16 @@ protected void doRun() throws Exception {
                 
                 String job_name;
                 Task task = item.task;
-                if (task instanceof FreeStyleProject){ 
+                if ((task instanceof FreeStyleProject) | (task instanceof WorkflowJob)) {
```
Comment (FlorianVeaux) [inline src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java:91]: Can the task really be of type WorkflowJob here ?

Comment (sarah-witt) [reply src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java:91]: `WorkflowJob` implements `Queue.Task`, so I wanted to be safe https://javadoc.jenkins.io/plugin/workflow-job/org/jenkinsci/plugins/workflow/job/WorkflowJob.html

Comment (FlorianVeaux) [reply src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java:91]: Have you seen that condition being true?
I'm afraid of the following:
- `getFullDisplayName` on WorkflowJob could return a build ID and cause context explosion.
- task could be an instance of both WorkflowJob and ExecutorStepExecution.PlaceholderTask leading to the wrong code being run

Comment (sarah-witt) [reply src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java:91]: I've seen it true in the tests but not in a jenkins instance. I can either get rid of this check or move it after the `ExecutorStepExecution.PlaceholderTask` check.

Comment (FlorianVeaux) [reply src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java:91]: > I can either get rid of this check

Seems safer for now until we find a real-life use case for something like this

**Pull request merged**